### PR TITLE
add(Script): IMDB Scraper

### LINKED ISSUE #311: IMDB_scrapper
**Is your feature request related to a problem? Please describe.**

IMDB_scrapper

**Describe the solution you'd like**

This script takes a movie name as input and returns the information about that movie.

### PR DESCRIPTION
**Pull request by priyanshu20**
closes #311 

## Modules Used

- requests
- bs4 (BeautifuSoup)
- argparse

<hr>

## How it works

- The User provides a query title and the script scrapes the IMDB website for data on it

- Since IMDB provides 200 results for every query, the data of top 5 is scraped, this is easily configurable.

## Usage

- Install dependencies
- python scraper.py --t movie-name-here
- sample : python scraper.py --t red
## Have you read the [Contributing Guidelines on Pull Requests](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/CONTRIBUTING.md)?

- [x] Yes
- [ ] No

## Type of change

_Please delete options that are not relevant._

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines(Clean Code) of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have created a helpful and easy to understand `README.md`
- [x] My documentation follows [`Template for README.md`](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/Template%20for%20README.md)
- [x] My changes generate no new warnings
- [x] I have added tests/screenshots(if any) that prove my fix is effective or that my feature works.

### CODE REVIEW

**Comment by santushtisharma10** [comment]
@kaustubhgupta @tamanna-verma @antrikshmisri

**Comment by antrikshmisri** [comment]
> @kaustubhgupta @tamanna-verma @antrikshmisri

Although there is a similar [script](https://github.com/avinashkranjan/Amazing-Python-Scripts/tree/master/Movie-Recommendation). I think this can be added

**Review by antrikshmisri — APPROVED** [review]

File: IMDB-Scraper/README.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+## How it works
+
+- The User provides a query title and the script scrapes the IMDB website for data on it
+
+- Since IMDB provides 200 results for every query, the data of top 5 is scraped, this is easily configurable.
+
+## Usage
+
+- Install dependencies
+- python scraper.py --t movie-name-here
```
Comment (kaustubhgupta) [inline IMDB-Scraper/README.md:23]:
```suggestion
- python scraper.py --t movie-name-here (in double quotes)
```

File: IMDB-Scraper/README.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+- sample : python scraper.py --t red
```
Comment (kaustubhgupta) [inline IMDB-Scraper/README.md:24]:
```suggestion
- sample : python scraper.py --t "red"
```

File: IMDB-Scraper/scraper.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
+import requests
```
Comment (kaustubhgupta) [inline IMDB-Scraper/scraper.py:1]: image
The output has several issues:
- Printing "Something went wrong" which should not be printed as it breaks the flow of reading.
- Look at Highlighted in yellow: it is printing non extra characters. (\xa0)
- You should not print entries which are not complete as it only floods the terminal. Only display values which are somewhat complete.

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]

File: IMDB-Scraper/scraper.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+# Base id url is used when the title id is known
+base_id = "https://www.imdb.com/title"
+# base url is used when the user gives a title to search for
+base = "https://www.imdb.com/find?s=tt&q="
+# for parsing we have used the lxml parser for optimization purposes, if lxml does not work for you replace 'lxml' with 'html.parser'
+
+
+def get_info(soup):
+    info = {}
```
Comment (antrikshmisri) [inline IMDB-Scraper/scraper.py:19]:
```suggestion
    info = {}
    labels = ['title', 'year', 'rating', 'genre', 'plot',
              'country', 'date', 'budget', 'gross_USA', 'gross_worldwide']
```

File: IMDB-Scraper/scraper.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+        blocks = details.findAll('div', attrs={"class": "txt-block"})
+      #   Strings have been stripped as the html of imdb is not clean
+      # Formatting of strings is preferrential you can change that to suit your needs
+        info["country"] = blocks[1].a.text
+        info["date"] = blocks[3].text[15:30]
+        info["budget"] = blocks[6].text[8:23]
+        info["gross_USA"] = blocks[8].text[11:]
+        info["gross_worldwide"] = blocks[9].text[28:]
+    except:
+        print("Something went wrong")
```
Comment (antrikshmisri) [inline IMDB-Scraper/scraper.py:41]:
```suggestion
        assert any(obj in labels for obj in info),"No info found"
```

**Review by antrikshmisri — CHANGES_REQUESTED** [review]
This should fix the `Something went wrong` exception being printed on each iteration.

**Comment by kaustubhgupta** [comment]
@antrikshmisri Good suggestions 💯

**Comment by priyanshu20** [comment]
Thank you everyone for your feedback I have made the necessary changes.
Requesting a re review from @kaustubhgupta  and @antrikshmisri

**Review by kaustubhgupta — APPROVED** [review]
Now it's good!

**Review by avinashkranjan — APPROVED** [review]

**Comment by avinashkranjan** [comment]
@all-contributors please add @priyanshu20 for code and documentation

**Pull request merged**